CONTRIBUTING.md

```diff
@@ -0,0 +1,219 @@
+our dependencies are already built by the community around conda.
+The remaining packages we maintain ourselves in our _ilastik-forge_ conda channel.
+
+Install miniconda:
+
+```bash
+# Install miniconda to the prefix of your choice, e.g. /home/<myuser>/miniconda3
+
+# LINUX:
+wget https://repo.continuum.io/miniconda/Miniconda3-latest-Linux-x86_64.sh
+bash Miniconda3-latest-Linux-x86_64.sh
+
+# MAC:
+wget https://repo.continuum.io/miniconda/Miniconda3-latest-MacOSX-x86_64.sh
+bash Miniconda3-latest-MacOSX-x86_64.sh
+
+# During the installation, conda edits your .bashrc to add its paths
+# Either re-open the terminal, or
+source ~/.bash_rc  # Linux
+source ~/.bash_profile  # MAC
+```
+
+Create the ilastik development environment (we assume that the dev-environment will have the name _idev_, but you can, of course choose a name to your liking):
+```bash
+conda create --name idev -c ilastik-forge -c conda-forge ilastik-dependenies-no-solvers
+```
+
+In order to run your own code, you need to link against your sources
+
+```bash
+# Define some variables that make the following lines more portable
+CONDA_ROOT=`conda info --root`
+DEV_PREFIX=${CONDA_ROOT}/envs/idev
+
+# first remote ilastik-meta from the conda environment
+conda --name idev remove ilastik-meta
+
+# now link against your own ilastik-meta repository
+# navigate to the idev environment root in your conda-folder
+cd ${DEV_PREFIX}
+ln -s <your_ilastik-meta_source_folder>  # e.g. ~/sources/ilastik-meta
+
+cat > ${DEV_PREFIX}/lib/python3.6/site-packages/ilastik-meta.pth << EOF
+../../../ilastik-meta/lazyflow
+../../../ilastik-meta/volumina
+../../../ilastik-meta/ilastik
+EOF
+```
+
+Now it might be the time to test whether the installation was successful:
+```bash
+# activate your conda environment
+source activate idev
+
+# check whether importing ilastik repos works
+python -c "import ilastik; import lazyflow; import volumina"
+```
+
+If there are no errors here, chances are high the development environment was set up correctly.
+If the above line already works, you can start ilastik by:
+
+```bash
+# in ~/source/ilastik-meta/ilastik
+python ilastik.py
+```
+
+More information on conda, and building the ilastik packages yourself:
+ * [miniconda](https://conda.io/miniconda.html)
+ * [conda user guide](https://conda.io/docs/user-guide/index.html)
+ * [conda build docs](https://conda.io/docs/commands/build/conda-build.html)
+ * [On building ilastik packages with conda-build](https://github.com/ilastik/ilastik-publish-packages)
+
+## Coding style
+
+Many users with different backgrounds have contributed to ilastik in the past.
+Code quality and coding styles can be quite different throughout the code-base.
+
+In general, when working on an existing file, please try to deduce the used coding style from what you see there,
+and adapt to it while working on this particular file.
```

Comment:
IMHO adapting the present style OR following the Google python style guide is both fine... Any thoughts?

Reply:
You mean for existing files?

Reply:
for me it's just easier to read if the same naming conventions are used in a single file, at least.

Reply:
Yes for existing files. We should aim at (slowly) converting to only one style though. I feel like forcing to keep an old style where it's present is counter productive... I see your point though, though one!

Reply:
Slowly converting implies that we will rewrite all code, eventually. To be honest, I don't see that happening. But I'm also not dogmatic about it. Let's collect more opinions and decide :)